enginetest/server_engine.go

```diff
@@ -72,74 +78,139 @@ func NewServerQueryEngine(t *testing.T, engine *sqle.Engine, builder server.Sess
+	if s.conn == nil {
+		err := s.NewConnection(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
 	// TODO
 	// q, bindVars, err := injectBindVarsAndPrepare(s.t, ctx, s.engine, query)
 	return nil, nil
 }
 
-func (s ServerQueryEngine) Query(ctx *sql.Context, query string) (sql.Schema, sql.RowIter, error) {
+func (s *ServerQueryEngine) Query(ctx *sql.Context, query string) (sql.Schema, sql.RowIter, error) {
+	if s.conn == nil {
+		err := s.NewConnection(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	// we prepare each query as prepared statement if possible to add more coverage to prepared tests
 	q, bindVars, err := injectBindVarsAndPrepare(s.t, ctx, s.engine, query)
 	if err != nil {
-		return nil, nil, err
+		// TODO: ctx being used does not get updated when running the queries through go sql driver.
+		//  we can try preparing and if it errors, then pass the original query
+		// For example, `USE db` does not change the db in the ctx.
+		return s.QueryWithBindings(ctx, query, nil, nil)
+		//return nil, nil, err
 	}
-
 	return s.QueryWithBindings(ctx, q, nil, bindVars)
 }
 
-func (s ServerQueryEngine) EngineAnalyzer() *analyzer.Analyzer {
+func (s *ServerQueryEngine) EngineAnalyzer() *analyzer.Analyzer {
 	return s.engine.Analyzer
 }
 
-func (s ServerQueryEngine) EnginePreparedDataCache() *sqle.PreparedDataCache {
+func (s *ServerQueryEngine) EnginePreparedDataCache() *sqle.PreparedDataCache {
 	return s.engine.PreparedDataCache
 }
 
-func (s ServerQueryEngine) QueryWithBindings(ctx *sql.Context, query string, parsed sqlparser.Statement, bindings map[string]*query.BindVariable) (sql.Schema, sql.RowIter, error) {
-	conn, err := newConnection(ctx)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	stmt, err := conn.Prepare(query)
-	if err != nil {
-		return nil, nil, err
+func (s *ServerQueryEngine) QueryWithBindings(ctx *sql.Context, query string, parsed sqlparser.Statement, bindings map[string]*query.BindVariable) (sql.Schema, sql.RowIter, error) {
+	if s.conn == nil {
+		err := s.NewConnection(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
 	}
 
-	args := prepareBindingArgs(bindings)
-
+	var err error
 	if parsed == nil {
 		parsed, err = sqlparser.Parse(query)
+		if err != nil {
+			// TODO: conn.Query() empty query does not error
+			if strings.HasSuffix(err.Error(), "empty statement") {
+				return nil, sql.RowsToRowIter(), nil
+			}
+			return nil, nil, err
+		}
+	}
+
+	// NOTE: MySQL does not support LOAD DATA query as PREPARED STATEMENT.
+	//  However, Dolt supports, but not go-sql-driver client
+	if _, ok := parsed.(*sqlparser.Load); ok {
+		rows, err := s.conn.Query(query)
 		if err != nil {
 			return nil, nil, err
 		}
+		return convertRowsResult(rows)
+	} else if _, ok := parsed.(*sqlparser.Execute); ok {
+		// TODO: cannot run `EXECUTE` query (need to replace it similar to how engine.go does)
+		return nil, sql.RowsToRowIter(), nil
+		//return s.engine.Query(ctx, query)
```

Comment:
Just wondering... maybe for these queries that we can't run, we should just call `t.Skip` so that we get some output that the tests wasn't really run fully.